### PR TITLE
Add all types to connections dialog and add icons

### LINKED ISSUE #60544: Connections dialog Extra Call Argument menu is missing many types
### Godot version

v4.0.alpha.custom_build [7b9fd0da6]

### System information

Windows 10

### Issue description

The connection dialog is missing many types. Other type selectors, including the array, dictionary, metadata, and project settings editors have all of the types.

| Connection Dialog | Array Editor |
---|---
| image | image |

### Steps to reproduce

* Select a node
* Switch to the Signals tab in the Node dock
* Double-click a signal to open the connection dialog
* In the bottom-right corner, click Advanced
* Click the dropdown button below "Add Extra Call Argument" in the top-right to see an incomplete list of types

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by fire-forge**
Closes https://github.com/godotengine/godot/issues/60544

Connections Dialog:
- Add all types to list (except Nil, Object, Callable, Signal, and RID) for parity with array, dictionary, metadata, and project settings editors.
- Add icons

| Before | After |
---|---
| image | image |

### CODE REVIEW

**Comment by akien-mga** [comment]
Did you test them all? I'm not sure how types like Object or Signal would be handled here.

**Comment by fire-forge** [comment]
I tested most of them. I agree that Object should probably be removed. I didn't remove the Signal and Callable types because of [this comment](https://github.com/godotengine/godot/pull/57175#issuecomment-1029176600) on a similar PR for the array and dictionary editors, but I can remove them if you think they should.

Signal and Callable are not handled well at all. Adding them doesn't work. The same issue exists for exported variables, and the array, dictionary, and project settings editors. I started a PR a while back to add EditorProperty classes for Callable and Signal that are similar to EditorPropertyRID, but I was running into issues with the scene file corrupting because Callable and Signal currently can't be serialized. 

I have an issue open for the array and dictionary editor specifically: https://github.com/godotengine/godot/issues/56916. I need to open an issue for Callable and Signal corrupting the scene file, which is an issue when exporting a Callable or Signal variable.

**Comment by fire-forge** [comment]
I just updated this PR to exclude Object, Callable, Signal, and RID types. Callable, Signal, and RID were recently removed from the array and dictionary editors in https://github.com/godotengine/godot/pull/60795.

File: editor/connections_dialog.cpp [hunk 1]
```diff
@@ -469,21 +429,14 @@ ConnectDialog::ConnectDialog() {
...
-	type_list->add_item("Vector2", Variant::VECTOR2);
-	type_list->add_item("Rect2", Variant::RECT2);
-	type_list->add_item("Vector3", Variant::VECTOR3);
-	type_list->add_item("Plane", Variant::PLANE);
-	type_list->add_item("Quaternion", Variant::QUATERNION);
-	type_list->add_item("AABB", Variant::AABB);
-	type_list->add_item("Basis", Variant::BASIS);
-	type_list->add_item("Transform3D", Variant::TRANSFORM3D);
-	type_list->add_item("Color", Variant::COLOR);
-	type_list->select(0);
```
Comment (akien-mga) [inline editor/connections_dialog.cpp:486]: Does it still pre-select something without this?

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**